flathunter/abstract_crawler.py

```diff
@@ -193,6 +196,71 @@ def resolve_geetest(self, driver):
             driver.refresh()
             raise
 
+    @backoff.on_exception(wait_gen=backoff.constant,
+                        exception=CaptchaUnsolvableError,
+                        max_tries=3)
+    def resolve_awsawf(self, driver):
+        """Resolve AWS WAF Captcha"""
+
+        # Intercept background network traffic via log sniffing
+        sleep(2)
+        logs_raw = driver.get_log("performance")
+        logs = [json.loads(lr["message"])["message"] for lr in logs_raw]
+        
+        def log_filter(log_):
+            return (
+                # is an actual response
+                log_["method"] == "Network.responseReceived"
+                # and json
+                and "json" in log_["params"]["response"]["mimeType"]
+            )
+        
+        for log in filter(log_filter, logs):
+            request_id = log["params"]["requestId"]
+            resp_url = log["params"]["response"]["url"]
+            if "problem" in resp_url and "awswaf" in resp_url:
+                response = driver.execute_cdp_cmd("Network.getResponseBody", {"requestId": request_id})
+                response_json = json.loads(response["body"])
+                iv = response_json["state"]["iv"]
+                context = response_json["state"]["payload"]
+                sitekey = response_json["key"]
+
+
+        sitekey = re.findall(
+            r"apiKey: \"(.*?)\"", driver.page_source)[0]
+
+        patternChallenge = r'src="([^"]*challenge\.js)"'
+        challenge_matches = re.findall(patternChallenge, driver.page_source)
+        for match in challenge_matches:
+            print(f'Challenge SRC Value: {match}')   
+            challenge = match
+
+        patternJsApi = r'src="([^"]*jsapi\.js)"'
+        jsapi_matches = re.findall(patternJsApi, driver.page_source)
+        for match in jsapi_matches:
+            print(f'JsApi SRC Value: {match}')   
```

Comment:
Please place `print` with `logger.debug`. Also you'll probably find for calls to the `logger` that the linter will a `%s` and a second argument to an f-string (because then it doesn't have to do the string interpolation if the log call isn't triggered).